File: TelerikUIExamples/TelerikUIExamples/DateTimeAxis.m
//
//  DateTimeAxis.m
//  TelerikUIExamples
//
//  Copyright (c) 2013 Telerik. All rights reserved.
//

#import "DateTimeAxis.h"
#import <TelerikUI/TelerikUI.h>

@implementation DateTimeAxis

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    TKChart *chart = [[TKChart alloc] initWithFrame:self.view.bounds];
    chart.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    [self.view addSubview:chart];
    
    NSCalendar *calendar = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierGregorian];
    NSDateComponents *dateTimeComponents = [[NSDateComponents alloc] init];
    dateTimeComponents.year = 2013;
    dateTimeComponents.day = 1;
    
    NSMutableArray *array = [[NSMutableArray alloc] init];
    for (int i = 1; i <= 6; i++) {
        dateTimeComponents.month = i;
        [array addObject:[[TKChartDataPoint alloc] initWithX:[calendar dateFromComponents:dateTimeComponents] Y:@(arc4random() % 100)]];
    }
    TKChartSplineAreaSeries *series = [[TKChartSplineAreaSeries alloc] initWithItems:array];
    series.selection = TKChartSeriesSelectionSeries;
    
    dateTimeComponents.month = 1;
    NSDate *minDate = [calendar dateFromComponents:dateTimeComponents];
    dateTimeComponents.month = 6;
    NSDate *maxDate = [calendar dateFromComponents:dateTimeComponents];
    
    // >> chart-axis-datetime
    TKChartDateTimeAxis *xAxis = [[TKChartDateTimeAxis alloc] initWithMinimumDate:minDate andMaximumDate:maxDate];
    xAxis.majorTickIntervalUnit = TKChartDateTimeAxisIntervalUnitMonths;
    xAxis.majorTickInterval = 1;
    // << chart-axis-datetime
    
    // >> chart-category-plot-onticks
    xAxis.plotMode = TKChartAxisPlotModeOnTicks;
    // << chart-category-plot-onticks
    
    chart.xAxis = xAxis;
    
    CGFloat shapeSize = ([UIDevice currentDevice].userInterfaceIdiom == UIUserInterfaceIdiomPhone) ? 10 : 17;
    series.style.pointShape = [[TKPredefinedShape alloc] initWithType:TKShapeTypeCircle andSize:CGSizeMake(shapeSize, shapeSize)];
    series.style.shapeMode = TKChartSeriesStyleShapeModeAlwaysShow;
    
    

    
    [chart addSeries:series];
    [chart reloadData];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/DateTimeAxis.swift
//
//  DateTimeAxis.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import Foundation

class DateTimeAxis:TKExamplesExampleViewController {
    
    let chart = TKChart()

    override func viewDidLoad() {
        super.viewDidLoad()
        
        chart.frame = self.view.bounds
        chart.autoresizingMask = UIViewAutoresizing(rawValue: UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        self.view.addSubview(chart)
        
        let calendar = NSCalendar(calendarIdentifier:NSCalendarIdentifierGregorian)!
        let dateTimeComponents = NSDateComponents()
        dateTimeComponents.year = 2013
        dateTimeComponents.day = 1
        
        var array = [TKChartDataPoint]()
        for i in 1..<7 {
            dateTimeComponents.month = i
            array.append(TKChartDataPoint(x:calendar.dateFromComponents(dateTimeComponents), y:Int(arc4random() % (100))))
        }
        
        let series = TKChartSplineAreaSeries(items:array)
        series.selection = TKChartSeriesSelection.Series
        
        dateTimeComponents.month = 1
        let minDate = calendar.dateFromComponents(dateTimeComponents)!
        dateTimeComponents.month = 6
        let maxDate = calendar.dateFromComponents(dateTimeComponents)!
    
        // >> chart-axis-datetime-swift
        let xAxis = TKChartDateTimeAxis(minimumDate: minDate, andMaximumDate: maxDate)
        xAxis.majorTickIntervalUnit = TKChartDateTimeAxisIntervalUnit.Months
        xAxis.majorTickInterval = 1
        // << chart-axis-datetime-swift
        
        // >> chart-category-plot-onticks-swift
        xAxis.setPlotMode(TKChartAxisPlotMode.OnTicks)
        // << chart-category-plot-onticks-swift
        
        chart.xAxis = xAxis
        
        chart.addSeries(series)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}